### PR TITLE
Travel time & distance calculator app

### LINKED ISSUE #3175: Application that calculates travel time and distance for OD pairs
Create a utility application that uses an osm.pbf file and a linkstats file and has a csv file with origin-destination coordinates as an input and writes a csv file with the same OD coordinates and calculated travel time and distance for them.

The input file columns
`id,origin_x,origin_y,destination_x,destination_y`

The output files should have the following columns:
`id,origin_x,origin_y,destination_x,destination_y,traveltime,distance`

We also need the following input parameters
1. router: R5 | GH (it should be R5 by default)
2. departure time: time in seconds when the vehicle departures (12 PM which is 12 * 3600 by default)

### PR DESCRIPTION
**Pull request by alex-vv**
Fixes #3175

As we discussed to skip Native CCH for now, I rebased my branch on top of develop so I can do a PR against the develop branch. 

This is for initial review as I want to investigate a minimal configuration for routers which takes osm.pbf only.

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3185)

### CODE REVIEW

File: src/main/scala/beam/router/TravelTimeAndDistanceCalculatorApp.scala [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+      graphDir = graphHopperDir,
+      geo = workerParams.geo,
+      vehicleTypes = workerParams.vehicleTypes,
+      fuelTypePrices = workerParams.fuelTypePrices,
+      wayId2TravelTime = wayId2TravelTime,
+      id2Link = id2Link
+    )
+  }
+
+  def parseArgs(args: Array[String]) = {
```
[the remark this answers is not in the document]
Comment (alex-vv) [reply src/main/scala/beam/router/TravelTimeAndDistanceCalculatorApp.scala:153]: ok, will use it

File: src/main/scala/beam/router/TravelTimeAndDistanceCalculatorApp.scala [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+  val geoUtils = new GeoUtils {
+    override def localCRS: String = "epsg:32631"
+  }
+
+  def processRow(row: CsvInputRow): CsvOutputRow = {
+    val streetVehicles: IndexedSeq[StreetVehicle] = Vector(
+      StreetVehicle(
+        Id.createVehicleId("1"),
+        Id.create("BODY-TYPE-DEFAULT", classOf[BeamVehicleType]),
```
[the remark this answers is not in the document]
Comment (alex-vv) [reply src/main/scala/beam/router/TravelTimeAndDistanceCalculatorApp.scala]: The problem is that it checks against type defined in vehicleTypes.csv. Ther only `BODY-TYPE-DEFAULT` is seems to be always available. In Beamville for example there is no `Car` only a custom `beamVilleCar` / `sharedCar` / `slowCar`

[the remark this answers is not in the document]
Comment (alex-vv) [reply src/main/scala/beam/router/TravelTimeAndDistanceCalculatorApp.scala]: Now I take the first vehicle type from available types which is a "car" category

**Comment by alex-vv** [comment]
Test!

**Review by JustinPihony — APPROVED** [review]

**Comment by JustinPihony** [comment]
> 00:06:27 TravelTimeAndDistanceCalculator
> 00:06:27 - should Run with R5 router *** FAILED *** (613 milliseconds)
> 00:06:27   java.io.FileNotFoundException: test/test-resources/beam/router/0.linkstats.csv.gz (No such file or directory)
> 00:06:27   at java.io.FileInputStream.open0(Native Method)
> 00:06:27   at java.io.FileInputStream.open(FileInputStream.java:195)
> 00:06:27   at java.io.FileInputStream.<init>(FileInputStream.java:138)
> 00:06:27   at java.io.FileInputStream.<init>(FileInputStream.java:93)
> 00:06:27   at beam.router.LinkTravelTimeContainer.loadLinkStats(LinkTravelTimeContainer.scala:28)
> 00:06:27   at beam.router.LinkTravelTimeContainer.<init>(LinkTravelTimeContainer.scala:21)
> 00:06:27   at beam.router.TravelTimeAndDistanceCalculatorApp.<init>(TravelTimeAndDistanceCalculatorApp.scala:94)
> 00:06:27   at beam.router.TravelTimeAndDistanceCalculatorAppSpec.$anonfun$new$2(TravelTimeAndDistanceCalculatorAppSpec.scala:21)
> 00:06:27   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
> 00:06:27   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)

**Comment by JustinPihony** [comment]
@alex-vv Do you still have the file as specified in 

`linkstatsPath = Paths.get("test/test-resources/beam/router/0.linkstats.csv.gz"),`

so that this test can run properly?

**Comment by alex-vv** [comment]
> @alex-vv Do you still have the file as specified in
> 
> `linkstatsPath = Paths.get("test/test-resources/beam/router/0.linkstats.csv.gz"),`
> 
> so that this test can run properly?

yes, sorry forgot to push it. Done now

**Comment by alex-vv** [comment]
Test!

**Comment by alex-vv** [comment]
test!

**Comment by alex-vv** [comment]
test!

**Comment by JustinPihony** [comment]
@alex-vv Is this ready to be merged?

**Comment by alex-vv** [comment]
@JustinPihony  yes, this is done

**Pull request merged**